apps/etl-func/src/adapters/cosmos/__tests__/cosmos-collector.test.ts

```diff
@@ -0,0 +1,138 @@
+import { Container } from "@azure/cosmos";
+import { pino } from "pino";
+import { beforeEach, describe, expect, test, vi } from "vitest";
+
+import { CosmosWeeklyEventCollector } from "../event-collector.js";
+
+const patchMock = vi.fn();
+const readMock = vi.fn();
+const createMock = vi.fn();
+
+const logger = pino();
+
+const containerMock = {
+  item: () => ({
+    read: readMock,
+  }),
+  items: {
+    create: createMock,
+  },
+} as unknown as Container;
+
+const cosmosCollectorMock = new CosmosWeeklyEventCollector(
+  containerMock,
+  logger,
+);
+
+const eventsMock = [{ foo: "bar" }];
+
+beforeEach(vi.resetAllMocks);
+
+describe("CosmosWeeklyEventCollector.collect", () => {
+  test("when the cosmos read fails it should thwor an error", async () => {
+    readMock.mockReturnValueOnce(
```

Comment:
Consider using [mockRejectedValueOnce](https://vitest.dev/api/mock#mockrejectedvalueonce) and its variant instead `mockReturnValue` or `mockImplementation`

Reply:
Thanks for the suggestion, take a look here [a3cb623](https://github.com/pagopa/io-messages/pull/85/commits/a3cb62366efc29589b16a25204d75b6d77be6566)